### DISCUSSION TITLE
Getting auto_quit to work

### DISCUSSION THREAD

**Discussion by lokesh-krishna** [R-nvim/R.nvim#91] [Q&A]
If I'm understanding the documentation right, R should auto quit by default when using the Neovim terminal. I went ahead and added `auto_quit = true` to my R.nvim configuration for good measure. However, quitting still doesn't quit the R console, unless I'm misunderstanding what this option is about. 

https://github.com/R-nvim/R.nvim/assets/20235646/d7bd5726-868f-4b5d-a4a0-09b1861c1972

It should have quit after I quit the file, right?

**Comment by jalvesaq** [comment]
Yes, this needs more testing. The `quit(save="no")` command is being sent to R, but I don't know if Neovim is waiting the specified maximum of 3 seconds to let R receive the command and process it (see lines 52-59 of `lua/r/edit.lua`).

**Comment by jalvesaq** [comment]
Note: not seeing the command being pasted doesn't mean that it wasn't pated because Neovim usually waits for the commands to be executed to redraw the user interface and, since it's quitting, maybe it doesn't redraw at all.

**Comment by jalvesaq** [comment]
To confirm that Neovim waits some time when `auto_quit` is `true` even when R is running in the built-in terminal emulator, I changed line 355 of `lua/r/run.lua` from

```lua
        qcmd = 'quit(save = "no")'
```

to

```lua
        qcmd = 'Sys.sleep(1); quit(save = "no")'
```

I can't see the command being pasted into the R Console, but I can note the extra delay to Neovim finishing. So, I think that this discussion can be closed if someone can confirm the same delay on macOS and Windows.
